contribs/gmf/src/directives/search.js

```diff
@@ -0,0 +1,281 @@
+/**
+ * @fileoverview this file provides the "search" directive and controller for
+ * a GeoMapFish application.
+ *
+ * Example:
+ *
+ * <gmf-search gmf-search-map="ctrl.map"></gmf-search>
+ */
+goog.provide('gmf.searchDirective');
+
+
+goog.require('gmf');
+goog.require('gmf.mapDirective');
+goog.require('ngeo.CreateGeoJSONBloodhound');
+goog.require('ngeo.searchDirective');
+goog.require('ol.Map');
+goog.require('ol.View');
+goog.require('ol.layer.Tile');
+goog.require('ol.layer.Vector');
+goog.require('ol.proj');
+goog.require('ol.source.OSM');
+goog.require('ol.source.Vector');
+
+
+/**
+ * @return {angular.Directive} The Directive Definition Object.
+ * @ngInject
+ */
+gmf.searchDirective = function() {
+  return {
+    restrict: 'E',
+    scope: {
+      'getMapFn': '&gmfSearchMap'
+    },
+    controller: 'GmfSearchController',
+    bindToController: true,
```

Comment:
I wouldn't use `bindToController: true` in this case, because you don't need to have `getMapFn` in the controller.